libs/template/file.go

```diff
@@ -68,13 +77,24 @@ func (f *copyFile) PersistToDisk() error {
 		return err
 	}
 	defer srcFile.Close()
-	dstFile, err := os.OpenFile(path, os.O_CREATE|os.O_EXCL|os.O_WRONLY, f.perm)
-	if err != nil {
+
+	ctx := context.Background()
```

Comment:
Please pass the context through the function args, and that of the caller. The reason for using `env.Lookup` on the context is that we can deal with environment overrides in the context instead of changing the actual environment variables. This means we can run parallel tests. Also, the workspace client may be passed through the context, which you default-construct here.

Reply:
done